### ISSUE TITLE
JUnit output contains escaped entities inside CDATA

### ISSUE THREAD

**Issue by joscha** [eslint/eslint#10537]
references https://github.com/eslint/eslint/pull/6672, https://github.com/bugcrowd/test-summary-buildkite-plugin/issues/19

When using the JUnit formatter, e.g. `eslint --format junit` then special characters like `"` are escaped within the output:

```xml
<testsuites>
	<testsuite package="org.eslint" time="0" tests="2" errors="2" name="/var/lib/buildkite-agent/builds/my-org/my-pipeline/web/bin/install_selenium.js">
		<testcase time="0" name="org.eslint.no-unused-vars">
			<failure message="'x' is assigned a value but never used.">
			<![CDATA[
			line 16, col 9, Error - &apos;x&apos; is assigned a value but never used. (no-unused-vars)
			]]>
			</failure>
		</testcase>
		<testcase time="0" name="org.eslint.semi">
			<failure message="Missing semicolon.">
			<![CDATA[ line 16, col 14, Error - Missing semicolon. (semi) ]]>
			</failure>
		</testcase>
	</testsuite>
</testsuites>
```

e.g.:
```
<![CDATA[
  line 16, col 9, Error - &apos;x&apos; is assigned a value but never used. (no-unused-vars)
]]>
```

however inside a CDATA section, entities (except for `]]>`) should not be escaped.

Thus I wonder if we can either:
* remove the CDATA section (anything coming from `xmlEscape` (https://github.com/eslint/eslint/blob/e862dc3a8e500afb54a229a20ce892272bf41860/lib/formatters/junit.js#L52) should be fine to just live inside a `<failure>` tag.
* remove the escaping (looking at https://github.com/eslint/eslint/pull/6672 we'd probably still need some way to transform escape characters into unicode characters)

**Tell us about your environment**

* **ESLint Version:** 4.16.0
* **Node Version:** 10.5.0
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

```

```bash
eslint --format junit
```

**What did you expect to happen?**

JUnit output CDATA sections to not have escaped characters.

**What actually happened? Please include the actual, raw output from ESLint.**

```bash
echo "const x = 'a';" > bla.js
yarn eslint bla.js --format junit
```
yields
```xml
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
<testsuite package="org.eslint" time="0" tests="1" errors="1" name="/Users/joscha/dev/bla.js">
<testcase time="0" name="org.eslint.no-unused-vars"><failure message="&apos;x&apos; is assigned a value but never used."><![CDATA[line 1, col 7, Error - &apos;x&apos; is assigned a value but never used. (no-unused-vars)]]></failure></testcase>
</testsuite>
</testsuites>
```

cc @tessereth

**Comment by joscha** [comment]
Dear lovely @eslint bot, it's a bug, I am pretty sure we can set the status to `accepted`, choose one of the solutions I proposed (or a different one) and I can make the changes.

**Comment by platinumazure** [comment]
Sorry @joscha, this one definitely slipped through the cracks.

I'll reopen and mark as "accepted" and "needs bikeshedding"-- that is, the bug is acknowledged to be real/reproducible but we need to work out minor design issues in the solution. And I'll see if I can get some more knowledgeable team members to weigh in on the right solution. Hope that sounds good :smile:

**Comment by joscha** [comment]
Thank you @platinumazure!
